### PR TITLE
[RHACS] ROX-21994: Migrating SCCs during the manual upgrade documentation

### PR DESCRIPTION
**Pull request by agantony**
Version(s):
4.4

Issue:
https://issues.redhat.com/browse/ROX-21994

Link to docs preview:
https://72486--docspreview.netlify.app/openshift-acs/latest/upgrading/upgrade-roxctl#migrating-sccs-during-the-manual-upgrade_upgrade-roxctl

SME review:
- [x] SME has approved this change.

Additional information:
- Merge into `rhacs-docs` and cherry-pick to:
    - `rhacs-docs-4.4`

### CODE REVIEW

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+<2> The name of the role resource.
+<3> The namespace in which the role is created. 
+<4> Describes the permissions granted by the role resource.
+<5> The type of Kubernetes resource, in this example, `RoleBinding`.
+<6> The name of the role binding resource.
+<7> Specifies the role to bind in the same namespace.
+<8> Specifies the subjects that are bound to the role.
+====
+
+.. Create the role and role binding resources specified in the `upgrade-scs.yaml` file by running the following command:
```
Comment (Maddosaurus) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:285]: We might want to tell customers they need to apply this file **to each secured cluster**.

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+[source,terminal]
+----
+$ oc delete scc/stackrox-central scc/stackrox-central-db scc/stackrox-scanner
+----
+
+* To delete the SCCs that are specific to all secured clusters, run the following command:
++
+[source,terminal]
+----
+$ oc delete scc/stackrox-admission-control scc/stackrox-collector scc/stackrox-sensor
```
Comment (Maddosaurus) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:305]: Same here, this command needs to be run on each cluster. I should have mentioned that in the initial document, apologies!

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+<6> The name of the role binding resource.
+<7> Specifies the role to bind in the same namespace.
+<8> Specifies the subjects that are bound to the role.
+====
+
+.. Create the role and role binding resources specified in the `update-central.yaml` file by running the following command:
++
+[source,terminal]
+----
+$ oc -n stackrox create -f ./update-central.yaml
```
Comment (Maddosaurus) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:213]: I don't know whether we should specify that this needs to be run only on the cluster that runs the central workload. It might be self-explanatory, so I trust you to know whether we need to add this here

**Review by Maddosaurus — APPROVED** [review]
Looks really good! I have left some comments where I could see us making the docs a bit clearer, but nothing blocking

**Comment by Maddosaurus** [comment]
Thanks for improving @agantony!

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 4]
```diff
@@ -0,0 +1,360 @@
+// Module included in the following assemblies:
+//
+// * upgrading/upgrade-roxctl.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="migrating-sccs-during-the-manual-upgrade_{context}"]
+= Migrating SCCs during the manual upgrade
+
+By migrating the security context constraints (SCCs) during the manual upgrade by using `roxctl` CLI, you can seamlessly transition the {rh-rhacs-first} services to use the stock {osp} SCCs, ensuring compatibility and optimal security configurations across Central and all secured clusters.
```
Comment (mburke5678) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:9]: Is _Central_ the full proper name? In some places in the assembly I see _Central database_. Wondering if this is the same or would a typical user (not me!) understand what we are referring to here?

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+
+.Procedure
+
+. List all the {product-title-short} services that are deployed on Central and all secured clusters:
```
Comment (mburke5678) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:13]:
```suggestion
. List all of the {product-title-short} services that are deployed on Central and all secured clusters:
```

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+Name:      scanner-68fc55b599-f2wm6
+           *openshift.io/scc: stackrox-scanner*
+Name:      scanner-68fc55b599-fztlh
+#...
+Name:      sensor-84545f86b7-xgdwf
+           *openshift.io/scc: stackrox-sensor*
+#...
+----
++
+In this example, you can see that each pod has its own custom SCC, which is specified through the `openshift.io/scc` field.
```
Comment (mburke5678) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:43]: Should we mention that the SCCs that we are concerned with here have `stackrox` in the name? Or, is that all to obvious?
```suggestion
In this example, you can see that each pod has its own custom SCC with `stackrox` in its name, which is specified through the `openshift.io/scc` field.
```

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+
+. Add the required roles and role bindings to use the stock {osp} SCCs instead of the {product-title-short} custom SCCs:
```
Comment (mburke5678) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:45]: I'm not clear on what is meant by _stock_ here and in the next step.  I wonder if we can remove it to avoid a potential translation issue?

```suggestion
. Add the required roles and role bindings to use the {osp} SCCs instead of the {product-title-short} custom SCCs:
```

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+====
+
+.. Create the role and role binding resources specified in the `update-central.yaml` file by running the following command:
++
+[source,terminal]
+----
+$ oc -n stackrox create -f ./update-central.yaml
+----
+
+* To add the required roles and role bindings to use the stock {osp} SCCs for all secured clusters, perform the following steps:
```
Comment (mburke5678) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:216]:
```suggestion
* To add the required roles and role bindings to use the {osp} SCCs for all secured clusters, perform the following steps:
```

**Comment by mburke5678** [comment]
@agantony A few suggestions, otherwise LGTM

File: modules/migrating-sccs-during-the-manual-upgrade.adoc (hunk 4 above)
Comment (agantony) [reply modules/migrating-sccs-during-the-manual-upgrade.adoc]: Yes, Central is the name of the component. No changes required.

File: modules/migrating-sccs-during-the-manual-upgrade.adoc (hunk 7 above)
Comment (Maddosaurus) [reply modules/migrating-sccs-during-the-manual-upgrade.adoc]: We are migrating from SCCs that we built ourselves (stackrox-*) to SCCs that ship with OpenShift (e.g. nonroot-v2), hence I called them "stock" SCCs. Feel free to rephrase this to ensure the instructions are clear!

File: modules/migrating-sccs-during-the-manual-upgrade.adoc (hunk 4 above)
Comment (Maddosaurus) [reply modules/migrating-sccs-during-the-manual-upgrade.adoc]: Technically, the full names for the two parts are `Central Services` and `Secured Cluster Services`. We often omit the `Services` part to shorten the name, but I trust you folks to decide whether we want to include the full name here or not.

Comment (agantony) [reply modules/migrating-sccs-during-the-manual-upgrade.adoc]: I would vote to leave it unchanged to maintain doc consistency.

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+Name:      scanner-68fc55b599-fztlh
+#...
+Name:      sensor-84545f86b7-xgdwf
+           *openshift.io/scc: stackrox-sensor*
+#...
+----
++
+In this example, you can see that each pod has its own custom SCC, which is specified through the `openshift.io/scc` field.
+
+. Add the required roles and role bindings to use the {osp} SCCs instead of the {product-title-short} custom SCCs.
```
Comment (kcarmichael08) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:45]: This technically includes the following 2 steps, so I would put it together with the next step. As it is now, it's a step with no actual content in it.

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+
+. To add the required roles and role bindings to the use the {osp} SCCs for the Central cluster, perform the following steps:
```
Comment (kcarmichael08) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:47]:
```suggestion
. To add the required roles and role bindings to use the {osp} SCCs for the Central cluster, perform the following steps:
```

File: modules/migrating-sccs-during-the-manual-upgrade.adoc (hunk 9 above)
Comment (kcarmichael08) [reply modules/migrating-sccs-during-the-manual-upgrade.adoc:45]: So just combine:

. Add the required roles and role bindings to use the {osp} SCCs instead of the {product-title-short} custom SCCs. To add the required roles and role bindings to use the {osp} SCCs for the Central cluster, perform the following steps:

**Review by kcarmichael08 — COMMENTED** [review]
Nice job! I had one suggestion and found one extra word to delete. 

A couple of other overall questions:

1. Does this apply only to RHACS for Red Hat OpenShift?
2. Is this optional or required? If required, should we add it as a bullet point in the list at the top of the page, before "Verify all the upgraded secured clusters"?

Depending on if it's for OpenShift only and/or required, we should add that info to the introductory paragraph before the first procedure.

**Comment by Maddosaurus** [comment]
@kcarmichael08 to answer your questions:

1. This applies to StackRox and RHACS installs on OpenShift only, as SCCs are an OpenShift exclusive feature at the time of writing.

2. This step is required if you have installed StackRox/RHACS *manually via manifests*. Helm installs and Operator installs will migrate SCCs automatically

File: modules/migrating-sccs-during-the-manual-upgrade.adoc [hunk 11]
```diff
@@ -0,0 +1,360 @@
+// Module included in the following assemblies:
+//
+// * upgrading/upgrade-roxctl.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="migrating-sccs-during-the-manual-upgrade_{context}"]
+= Migrating SCCs during the manual upgrade
+
+By migrating the security context constraints (SCCs) during the manual upgrade by using `roxctl` CLI, you can seamlessly transition the {rh-rhacs-first} services to use the {osp} SCCs, ensuring compatibility and optimal security configurations across Central and all secured clusters.
+
```
Comment (kcarmichael08) [inline modules/migrating-sccs-during-the-manual-upgrade.adoc:10]: could we add a sentence here? 

This step is required when upgrading to release 4.4 if you installed {product-title-short} for {osp} manually by using the manifest installation method.

**Review by kcarmichael08 — COMMENTED** [review]
One question and we need to squash, otherwise it's good to merge!
